lib/charms/opensearch/v0/opensearch_relation_provider.py

```diff
@@ -343,8 +354,9 @@ def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
         if not self.unit.is_leader():
             return
         if self._unit_departing(event.relation):
-            # This unit is being removed, so don't update the relation.
+            # This unit is being removed.
```

Comment:
is this supposed to be ran only on the leader?

Reply:
Update_endpoints only runs on the leader because it updates the app databag. However, I've moved this function to the departing hook, since we can directly omit the `event.departing_unit`.